server/src/main/java/io/crate/action/sql/Cursor.java

```diff
@@ -235,6 +231,16 @@ private void resetCursorToMaxBufferedRowsPlus1() {
         }
     }
 
+    private int newCursorPosition(int count) {
+        int newCursorPosition = cursorPosition + count;
+        // HD 2-12 Overflow iff both arguments have the opposite sign of the result
+        if (((cursorPosition ^ newCursorPosition) & (count ^ newCursorPosition)) < 0) {
```

Comment:
minor: maybe we can mention in the code comments. that it's taken from `Math.add`.